File: ClientApp/src/app/services/wizard-step.service.ts
import { Injectable, Inject } from '@angular/core';
import { BaseDataFetchService } from './base-data-fetch-service';
import { HttpClient } from '@angular/common/http';
import { WizardStep } from '../models/wizard-step.model';
import { Observable } from 'rxjs';

@Injectable()
export class WizardStepService extends BaseDataFetchService {

    private path: string;

    constructor(http: HttpClient, @Inject('BASE_URL') BASE_URL: string) {
        super(http, BASE_URL);
        this.path = `${this.baseUrl}` + 'WizardStep/';
    }

    getStep(stepId: string) {
        return this.get<WizardStep>(this.path, stepId);
    }

    getSteps() {
        return this.getAll<WizardStep>(this.path);
    }

    createStep(title) {
        return this.create<WizardStep>(this.path, { Title: title });
    }

    deleteStep(stepId: string) {
        return this.delete<any>(this.path, stepId);
    }
}

File: Services/WizardStepService.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using WizardLayout.Common.Interfaces;
using WizardLayout.Common.Interfaces.Services;
using WizardLayout.Model;

namespace WizardLayout.Services
{
    public class WizardStepService : IWizardStepService
    {
        protected IWizardStepRepository _wizardStepRepository { get; }
        public WizardStepService(IWizardStepRepository wizardStepRepository)
        {
            _wizardStepRepository = wizardStepRepository;
        }

        public Task<WizardStep> CreateStep(WizardStep wizardStep) => _wizardStepRepository.Add(wizardStep);
        public Task<WizardStep> DeleteStep(int stepId) => _wizardStepRepository.Delete(stepId);
        public Task<WizardStep> GetStep(int stepId) => _wizardStepRepository.Get(stepId);
        public Task<List<WizardStep>> GetSteps(int? pageIndex, int? pageSize) => _wizardStepRepository.GetAll(pageIndex, pageSize);

    }
}
